python/social_bot/gazebo_agent.py

```diff
@@ -180,10 +195,37 @@ def get_full_states_observation(self, teacher):
         """
         task_specific_ob = teacher.get_task_specific_observation(self)
         agent_pose = np.array(self.get_pose()).flatten()
-        agent_vel = np.array(self.get_velocities()).flatten()
-        internal_states = self.get_internal_states()
-        obs = np.concatenate(
-            (task_specific_ob, agent_pose, agent_vel, internal_states), axis=0)
+        if self._use_simple_full_states:
+            # assumes GoalTask and that the first 3 dims of the
+            # task_specific_observation give the goal position.
+            yaw = agent_pose[5]
+            obs = []
+            while len(task_specific_ob) > 1:
+                x = task_specific_ob[0] - agent_pose[0]
```

Comment:
Perhaps use angle directly would be a little bit simpler?
```
        object_direction = np.arctan2(object_pos[1] - agent_pos[1], object_pos[0] - agent_pos[0])
        view_angle = object_direction - yaw
        if view_angle > self._view_angle_limit or view_angle < -self._view_angle_limit:
                view_angle = 0
                obs.extend([view_angle, 0])
        else: 
                obs.extend([view_angle, 1])
```

Reply:
Good point.  I thought about that, but the problem with angle is it's discontinuous.  Coordinates also convey a sense of depth, that's why I'm using rotated_x and _y.

Reply:
The code I wrote above is discontinuous...Sorry. We can first translate (x,y) to agent's coordinate, and then do the arctan2. So we have a range of arctan2 (-pi, pi).

If view_angle_limit < pi, the angle's range is (- view_angle_limit, view_angle_limit), there is no discontinuous problem.
If view_angle_limit >= pi, the range (-pi, pi). it is discontinuous when the object is behind the agent by 180 degree. But that's fine because the optimal action for the states is actually discontinuous: near to -pi, the agent should turn left, near to pi, the agent should turn right. This is a more "actionable" representation than coordinates.